### ISSUE TITLE
Not able to ompile soxr library for Apple Silicon M1 Pro

### ISSUE THREAD

**Issue by flak228** [conan-io/conan-center-index#23118]
### What is your question?

Hey!)
I want to mention, that I'm a newbee in programming, so I'm really sorry, if something there is sound dumb, I just trying to do my best with it!)

I need to compile soxr library to use it with my FFMPEG build for M1 Pro
If I understood corredtly, it doesn't support Apple Silicon for now, so I got to this brunch, and tried to use that fix!)
I download conandata.yml, conanfile.py and patches form the root folder, and t
hem put them in the main folder
Then I used command conan install .
And then I used command conan create . --version=0.1.3
And then, according to the instructions, I run ./conanbuild.sh
And after that I had new file, deactivate_conanbuild-env-release-armv8.sh
And nothing more(
And if I got it right, for my purposes I need to have file with .a to use it in the next steps!)
Can I ask oyu to help me to understand what I suppose to do with that?)

One more time wants to say big sorry for being not so good in it, and gonna be really thankful for any help!)
![image_conan](https://github.com/conan-io/conan-center-index/assets/22274250/d4a1a9bc-106c-4956-a205-a19ebcdc328e)

**Comment by uilianries** [comment]
@flak228 Thank you for sharing your scenario!

> I need to compile soxr library to use it with my FFMPEG build for M1 Pro

Indeed `soxr` is not managed right now in the ffmpeg recipe, it would require a new PR adding a new option, configuring cpp_info and managing the option by its dependencies. The ffmpeg is not an easy Conan recipe to understand, as you mentioned be a newbie, you may have difficult to understand that recipe at first sight.

>  If I understood corredtly, it doesn't support Apple Silicon for now, so I got to this brunch, and tried to use that fix!)

Actually, we have the [soxr](https://conan.io/center/recipes/soxr?version=0.1.3) library packaged already in Conan Center and it has Mac M1 packages available (you can see on the Packages tab). Please, try that package first.

> I download conandata.yml, conanfile.py and patches form the root folder, and them put them in the main folder

I'm not sure which guide you are using, but I would recommend you just cloning conan-center-index and building what you need instead. I expect you are running Conan 2.6.0 or similar version:

```
git clone https://github.com/conan-io/conan-center-index.git
cd conan-center-index/recipes/soxr/
conan create all/ --version=0.1.3
```

Another way, is using the Conan install command, so you do not need to clone the repository. Conan client will download the recipe and build from source automatically:

    conan install -r conancenter --requires=soxr/0.1.3 --build=soxr/0.1.3
    
This command says to Conan install `soxr/0.1.3` from the official Conan repository `conancenter`, and to build `soxr/0.1.3` from source. 

> Can I ask oyu to help me to understand what I suppose to do with that?)

You could submit a PR to Conan Center Index. As reference, for your particular case, you can use these PRs as reference, they are changing ffmpeg to add a new option related to a dependency:

- https://github.com/conan-io/conan-center-index/pull/24879
- https://github.com/conan-io/conan-center-index/pull/22990/files#diff-13ca8667b24368a779b16b2c336319a64d00888f52929213802d3572e58828bcR333

Don't worry you get blocked by an error, you can always ask for help in your PR. A reviewer from Conan community should help you.

Regards.

**Comment by AbrilRBS** [comment]
Hi @flak228 thanks a lot for your issue

We have taken the chance to implement your request of soxr in ffmpeg after confirming that soxr compiles successfuly locally for Macos M2. Is soxr compilation still an issue for your?

You can try locally with `conan install --requires=soxr/0.1.3 -u -b=missing`. If this fails, sharing the whole compilation logs could be very useful to track down the issue, thanks!

If soxr is not an issue anymore, we'll close this issue after merging the PR linked above. Thanks again for your patience :)
